docs/concepts/native-interop.rst

```diff
@@ -0,0 +1,260 @@
+Native interoperability
+=======================
+
+In this document we will dive a little bit deeper into all three ways of doing 
+"native interoperability" that are available on the .NET platform. 
+
+To refresh your memory, the two main ways of doing native interoperability are:
+
+* Platform Invoke (P/Invoke)
+* COM interoperability
```

Comment:
Saying that COM Interop is available on Windows isn't quite clear enough. It's not a .NET Core feature (to my knowledge), but a .NET Framework one. As a result, I don't consider it a .NET Platform feature. It's not documented in ECMA.

It's fine to document it, but I'd just put it in a .NET Framework specific document and link to it.

Miguel gave us the advice to approach the platform with a Unix-first mentality. I think that we need to apply that guidance as best we can.